add dub filtering for personal animelist

Funami580 opened the pull request:
In case you are interested in that functionality. I built it into the [German version of this project](https://github.com/Funami580/MAL-GerDubs). And thanks for creating this useful project!

MAL-Dubs commented:
Wow! I'm so flattered that you'd fork this. For now I’ve decided not to implement this change as I'm refactoring a few things to add other new features, but I will keep this in mind.

The main thing that kept me from adding filtering to user lists was custom list CSS and the classic template. It's like trying to hit a moving target figuring out where to put it 😓

Testing this made me recognize the poor performance of updating the class of every filtered element every time they're shown/hidden. To fix this I updated the filter to mark container elements once, on load, and show or hide them using a single class change on the `body` 1e524ca5707c3f7fcba6e29c1c440fc24f409106. I also changed the Dub Only checkbox to use Font Awesome to match the seasonal page 86649cbfa4b3f06601be9426571e6ba008f76ff8. This font-based checkbox would be easier to style on different list backgrounds.

An unfortunate side effect of these changes is that they broke the checkbox and filtering on your version. The `@resource CSS` in your script currently links to this repository’s css/style.css. You should change that line in your script to the file in your repo as well as the feedback links in the menu and any other dependencies.

Your scraper is amazing. Maybe at some point it would make sense to add that data and give users a language preference. Compiling and keeping the data up-to-date and accurate has been the most involved aspect of this project, and automating it for different languages could make that possible. Really impressive work.

Thank you for your contribution and your kind words about the project! 🙏

Funami580 commented:
> Wow! I'm so flattered that you'd fork this. For now I’ve decided not to implement this change as I'm refactoring a few things to add other new features, but I will keep this in mind.

Thanks a lot for your really kind words! :)

> The main thing that kept me from adding filtering to user lists was custom list CSS and the classic template. It's like trying to hit a moving target figuring out where to put it 😓

True, I don't use custom list CSS myself so I went ahead and implemented it anyway. ^^

> Testing this made me recognize the poor performance of updating the class of every filtered element every time they're shown/hidden. To fix this I updated the filter to mark container elements once, on load, and show or hide them using a single class change on the `body` [1e524ca](https://github.com/MAL-Dubs/MAL-Dubs/commit/1e524ca5707c3f7fcba6e29c1c440fc24f409106). I also changed the Dub Only checkbox to use Font Awesome to match the seasonal page [86649cb](https://github.com/MAL-Dubs/MAL-Dubs/commit/86649cbfa4b3f06601be9426571e6ba008f76ff8). This font-based checkbox would be easier to style on different list backgrounds.

Good idea! I implemented the changes in my fork.

> An unfortunate side effect of these changes is that they broke the checkbox and filtering on your version. The `@resource CSS` in your script currently links to this repository’s css/style.css. You should change that line in your script to the file in your repo as well as the feedback links in the menu and any other dependencies.

Yes, you're right. I changed that just now. ^^

> Your scraper is amazing. Maybe at some point it would make sense to add that data and give users a language preference.

That's a cool idea. Via this variable it's even possible to scrape data for other languages than German: [gen_data/src/anisearch.rs#L6](https://github.com/Funami580/MAL-GerDubs/blob/3c233c348db44fce43f26ffcb5e21e6bf6e9c350/gen_data/src/anisearch.rs#L6) (Theoretically, I could also make it changeable via the console.)
You could change it for example to `en` (English), `fr` (French), `it` (Italian), and `es` (Spanish). But in the end you would have to rely on aniSearch, and I don't know how complete and up-to-date their data is. For example, they don't show [Semiwa Magic Cube](https://www.anisearch.com/anime/13780,semiwa-magic-cube) with an English dub yet.

> Compiling and keeping the data up-to-date and accurate has been the most involved aspect of this project, and automating it for different languages could make that possible. Really impressive work.

Yes, I can only imagine. A scraper is written kind of fast, but being dedicated to update the data by hand is what I find impressive!

> Thank you for your contribution and your kind words about the project! 🙏

And thank you for creating this project and keeping it alive! :)
